homework-submission/blog/src/App.js

```diff
@@ -0,0 +1,85 @@
+import React, { Component } from 'react';
+import Comments from './Comments';
+import './App.css';
+import AddForm from './AddForm';
+
+class App extends Component {
+  state = {
+    data: [],
+  };
+  componentDidMount() {
+    this.reset();
+  }
+
+  reset = () => {
+    fetch('https://hyf-react-api.herokuapp.com/blog/comments')
+      .then(response => response.json())
+      .then(data => this.setState({ data }));
+  };
+
+  deleteComment = id => {
+    fetch(`https://hyf-react-api.herokuapp.com/blog/comments/${id}`, {
+      method: 'DELETE',
+    })
+      .then(response => response.json())
+      .then(response => {
+        let newData = this.state.data.filter(comment => response._id !== comment._id);
+        this.setState({ data: newData });
+      });
+  };
+
+  addComment = comment => {
+    fetch('https://hyf-react-api.herokuapp.com/blog/comments/create', {
+      method: 'POST',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(comment),
+    })
+      .then(comment => comment.json())
+      .then(comment => {
+        this.setState({ data: [...this.state.data, comment] });
+      });
+  };
+
+  handleUpdate = (updatedItem, id) => {
+    fetch(`https://hyf-react-api.herokuapp.com/blog/comments/${id}`, {
+      method: 'PATCH',
+      headers: {
+        'Content-Type': 'application/json',
+      },
+      body: JSON.stringify(updatedItem),
+    })
+      .then(response => response.json())
+      .then(response => {
+        this.state.data.map(comment => {
+          if (comment._id === response._id) {
+            response.text = comment.text;
+            response.date = comment.date;
+            response.author.firstName = comment.author.firstName;
+            response.author.lastName = comment.author.lastName;
+          }
+          return comment;
+        });
+        this.reset();
```

Comment:
you are fetching the all data, just for one updated item and that's not good
try to find another solution